### DISCUSSION TITLE
What does "Fill" do?

### DISCUSSION THREAD

**Discussion by realhackcraft** [rxhanson/RectanglePro-Community#261] [Q&A] [answered]
I can't figure out the difference between fill left and left half. They do the same when I test it.

**Comment by rxhanson** [comment, marked as answer]
Fill left/right will take other visible windows into account, if there is sufficient space available next to other windows. If there is no other window visible or if there is insufficient space, then fill left/right will execute left/right half. 

Repeat fill commands to cycle through all of the potential sizes.

**Reply by realhackcraft** [reply]
Please help me

**Comment by realhackcraft** [comment]
I still don't understand how this works.

here is a video:
https://user-images.githubusercontent.com/65378120/233185323-7b2b667e-395f-40f3-b40d-038a0b501346.mov

**Comment by realhackcraft** [comment]
Even when there is load of space, it still executes right half

https://user-images.githubusercontent.com/65378120/233186005-889b1404-eb06-42c2-81bf-90abf48526fc.mov

**Comment by rxhanson** [comment]
Thanks for the screen recordings. I think the misunderstanding here is that fill right is for filling any area remaining to the right of a window, not below a window. I can see where this would be confusing. The use-case that fill left/right is solving is more about having two windows taking up the left and right halves but not being exactly 1/2 the screen. Try moving a window anywhere to the left side of the screen, and then doing a fill right on another window and perhaps it will help communicate what these currently do.

**Reply by masont** [reply]
Hi, I just downloaded Rectangle Pro and considering to upgrade from Rectangle. The "Fill" functionality would be the main benefit for me and reason enough to upgrade. I'm running into what I think is this behavior which I find surprising. Basically it seems like the "sufficient space" heuristic isn't matching my use case, which is that I want the Fill to always just fill the available space taking into account any visible windows, even if it's rather small.

Specifically, I have an ultrawide monitor and often have a Zoom meeting in the middle third-ish (not exactly third). I want to be able to quickly move a document window in Chrome to the remaining space on the left. Resizing to half-ish doesn't work, and it seems like because the remaining space is small enough, Rectangle Pro keeps defaulting to a different window on the screen for calculating the remaining horizontal space.
